### DISCUSSION TITLE
SQLRDD + MT

### DISCUSSION THREAD

**Discussion by sygecom** [marcosgambeta/sqlrddpp#15] [General]
Marcos,
Ficou muito boa essa versão para Harbour, sabe me dizer se posso rodar com -MT e se precisa adicionar algo na hora de compilar a LIB para usar com -MT ?

**Comment by prgwiz** [comment]
Each thread should open up the tables it needs.  There were a couple static
variables in sqlrdd that I modified to be thread safe but you may not need
to.
I have tested MT with importing dbf's into Postgres.  I have it working now.

**Reply by sygecom** [reply]
Thanks for the feedback.
In this case I am not using wordarea.
I am using direct SQL command.
But I get an error below:

Assertion failed: res != NULL, file ../../../../../../source/pgs.c, line 131

Could you give me any tips?

**Comment by prgwiz** [comment]
Did you reconnect in the thread?

**Reply by sygecom** [reply]
I didn't understand.

For each FUNCTION that has several SQL calls, there is a thread.

For each SQL I connect to the database, I execute the SQL and close the connection.

**Comment by sygecom** [comment]
I've done everything, but I get an error in POSTGRESQL, it closes my application out of nowhere and records the following error in the POSTGRESQL LOG:

could not receive data from client: An existing connection was forcibly closed by the remote host.

**Comment by prgwiz** [comment]
I am using sqlrdd and at the end of the thread routine I have:
oSql:=SR_getconnection(nConnection)
oSql:end()

**Reply by sygecom** [reply]
I also tried and still got an error.

One question, when compiling the LIB do I need to put it in: sqlrddpp.hbp -mt ?

Or just on my system?

**Comment by prgwiz** [comment]
Just on your system.
Take a look at /SQLRDD/source/sqlrdd0.prg
     static aConnections, nActiveConnection
You have to make it  thread safe/aware and anything that references them.
I made a function to get/set and they default to the values per value of
hb_threadid().

**Reply by sygecom** [reply]
Thanks for the feedback.
Could you share your improvements so I can try here?
